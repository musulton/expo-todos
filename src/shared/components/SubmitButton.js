import {TouchableHighlight, Text, StyleSheet, View} from "react-native";
import {getAccessibilityLabel} from "../utils/Functions";

const SubmitButton = ({onSubmit, title}) => (
    <View style={styles.container}>
        <TouchableHighlight
            underlayColor={"#efefef"}
            onPress={onSubmit}
            style={styles.button}
            {...getAccessibilityLabel("Submit_Button_Component")}
        >
            <Text style={styles.text}>{title}</Text>
        </TouchableHighlight>
    </View>
)

export default SubmitButton;

const styles = StyleSheet.create({
    container: {
        alignSelf: "flex-end",
        marginVertical: 10
    },
    button: {
        height: 50,
        paddingHorizontal: 20,
        width: 150,
        borderWidth: 1,
        borderColor: "rgba(0,0,0, 0.1)",
        justifyContent: "center",
        alignItems: "center"
    },
    text: {
        color: "#666666",
        fontWeight: "bold"
    }
})
