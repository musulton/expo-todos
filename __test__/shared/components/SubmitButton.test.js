import {render, fireEvent, screen} from "@testing-library/react-native";

import SubmitButton from "../../../src/shared/components/SubmitButton";

describe("SubmitButton", () => {
    const props = {
        title: "Title",
        onSubmit: jest.fn()
    };

    test("should call onSubmit props when button is clicked", () => {
        render(<SubmitButton {...props} />);
        fireEvent.press(screen.getByText(props.title));

        expect(props.onSubmit).toHaveBeenCalled();
    })

    test("should render correctly", () => {
        render(<SubmitButton {...props} />);

        expect(screen.toJSON()).toMatchSnapshot();
    })
})
