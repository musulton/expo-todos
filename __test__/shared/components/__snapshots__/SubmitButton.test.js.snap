// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly 1`] = `
<View
  style={
    {
      "alignSelf": "flex-end",
      "marginVertical": 10,
    }
  }
>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "rgba(0,0,0, 0.1)",
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": 150,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666666",
          "fontWeight": "bold",
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
